src/XRP/issued-currency-client.ts

```diff
@@ -379,4 +385,163 @@ export default class IssuedCurrencyClient {
       wallet,
     )
   }
+
+  /**
+   * Sends issued currency from one account to another.  This method can be used to create issued currency, dispense issued currency from
+   * an operational address, send issued currency from one XRPL account to another (as long as the payment only involves a single currency,
+   * i.e. is not cross-currency), or to redeem issued currency at the issuing address.
+   * The specific case depends on the relationship among the parameters.
+   *
+   * TODO: (acorso) Remove this, and make that ^^ better.: ================================================================================
+   * Notes to reviewers:
+   *
+   * Case - ALL:
+   *      The destination address is NOT the same as the sending address of the transaction.
+   *      (This should only be true for currency exchange payments, which is a cross-currency payment to yourself.)
+   *
+   *      The SendMax field (if present) should always indicate the same currency and issuer as the Amount field... otherwise, it would
+   *      be considered a cross-currency payment (which are triggered by different currencies in Amount v.s. SendMax fields).
+   *      SendMax values are not parameterized here, and are instead derived from the provided values for issuer, currency, etc.
+   *       - therefore this method is incapable of initiating a cross-currency payment.
+   *
+   *
+   * Case - Creating an issued currency:
+   *      The sending address of the transaction is the same address as the issuer of the `Amount` field.
+   *
+   *      No SendMax field is required because the Transfer Fee doesn't apply to transactions in which
+   *              the originator or the destination address is the same as the issuer.
+   *               (you don't get charged to handle your own money, or let your customers redeem their money with you.)
+   *
+   *      ... does the destination address need to have a trustline established to the sender/issuer?
+   *
+   *
+   * Case - dispensing an issued currency from an operational address (https://xrpl.org/issuing-and-operational-addresses.html):
+   *      The sending address (an operational address) is not the same address as the issuer of the `Amount` field
+   *
+   *      Issuer needs to allow rippling (https://xrpl.org/rippling.html).
+   *
+   *      A SendMax field is required here, because the transfer fee will apply as the sender of this txn is not the issuer.
+   *
+   *
+   * Case - sending an issued currency payment from one XRPL address to another:
+   *      Identical to dispensing an issued currency from an operational address (which is just making an issued currency payment)
+   *
+   *
+   * Case - redeeming an issued currency with the issuer:
+   *      The destination address is the same address as the issuer of the `Amount` field.
```

Comment:
> The specific case depends on the relationship among the parameters.

this is pretty intense

the downside with this is you have to rely on documentation and your knowledge of how the parameters interact in order to get things done, as opposed to a more opinionated api.

i can't tell yet if we actually want more abstract apis for each of these functions (i feel like if i were building a DEX or gateway that used this stuff, i would want them), but either way we would probably want a lower-level escape hatch so it seems reasonable for this function to be public.

if you *really* wanted the mega function, one way you could enforce some exclusivity with the type system would be to do something like

```
interface CaseOne {
  destination: string,
  transferFee: number,
}

interface CaseTwo {
  destination: string,
  mutuallyExclusiveField: number
}

type SendIssuedCurrencyOptions = CaseOne | CaseTwo

function sendIssuedCurrency(options: SendIssuedCurrencyOptions) ...
```

not sure if there's a sufficient number of combinations for this to be useful, though.

in the future, it might be good to make PRs that focus on the individual cases instead of doing it all at once. there might be some overlap in what earlier PRs implement, but it's easier to follow

Reply:
So the notes on this method definitely make it seem particularly intense... (I plan to remove these soon enough).  I think we may want to expose a more opinionated API, because that will help make it clear what information needs to be provided for each case.  For example, if you want to issue some new issued currency, the API simply won't ask you for the transfer fee, because it does not apply in this case.  Similarly, if you want to redeem some issued currency, it simply won't ask you for the destination, because it'll use the same address as for `issuer` as the destination and error if there's a problem.

I think it may make sense to leave this private... but if we are able to expose all possible parameters that one *might* want to use for each distinct use case, I think we could make this the private workhorse underpinning all of those cases.

Reply:
cool, that sounds reasonable to me. will comment on the naming possibilties in slack